### PR TITLE
[4.0] Impossible to Prepare multilingual site associations with Content Language when site language is not yet installed

### PR DESCRIPTION
**Pull request by infograf768**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/27236#issuecomment-568279822

### Summary of Changes
Since the beginning of the multilingual feature in Joomla, it was designed to be able to prepare a site with a Content Language even if the site corresponding language was not yet installed.

This has been broken by forcing `$checkInstalled` to be true in https://github.com/joomla/joomla-cms/blob/4.0-dev/libraries/src/Language/LanguageHelper.php#L365 (3.7.0)
and using `true` in other places where the `getContentLanguages()` method is used.

It may have been necessary at the time, but afaik it is useless now and breaks the feature.

This PR sets `$checkInstalled` to `false` by default and modifies to `false` the instances where `true` is used.

### Testing Instructions
Install a multingual site with the multilingual sample data.
Create a new Content Language without its site Language. Let's use for example pt-PT for the demo
Create category and content (article, menus, categories, newsfeeds, contacts), tagged to the Content Language: no problem with that.

Edit an item NOT tagged to the pt-PT Content Language, switch to the Associations tab.
**pt-PT will not display.**
Display the Multilingual Associations component: same result, pt-PT will not display.

Patch and test again

After patch, test that all works fine in frontend. pt-PT should not display in the module switcher, etc.

### After patch
All goes fine.
Example: Portuguese (pt-PT) displays fine in Associations Tab:
Screen Shot 2019-12-23 at 11 06 55

### Documentation Changes Required
It may need a change indeed if there is a doc about it...

@Fedik

### CODE REVIEW

**Comment by Fedik** [comment]
thanks for fix, I will test this evening

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 5c9d2d5efc308b5caa264cf16b3e958de53d21ca<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27341">issues.joomla.org/tracker/joomla-cms/27341</a>.</sub>

**Comment by infograf768** [comment]
**Note:** `com_mails` may be considered as a specific case where `true` has to be used if the code remains as is... But the original creator of this component refused to accept my input.

I still think that `getContentLanguages()` is the wrong method to use there if the site is not multilingual. 
It should use `getInstalledLanguages()` and not display flags but badges with the language tag.

If the site is multilingual, it could use `getContentLanguages()` with `true` added i.e. making sure the corresponding language is installed.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 17c6a5e202b0ab38b4286af5d94eb5d4c9a29046<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27341">issues.joomla.org/tracker/joomla-cms/27341</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27341">issues.joomla.org/tracker/joomla-cms/27341</a>.</sub>

**Comment by richard67** [comment]
Test result of Fedik was still valid because commits after his test were just updates to 4.0-dev without any changes on this PR.

File: libraries/src/Language/LanguageHelper.php [hunk 1]
```diff
@@ -362,7 +362,7 @@ public static function getInstalledLanguages($clientId = null, $processMetaData
 	 *
 	 * @since   3.7.0
 	 */
-	public static function getContentLanguages($publishedStates = array(1), $checkInstalled = true, $pivot = 'lang_code', $orderField = null,
+	public static function getContentLanguages($publishedStates = array(1), $checkInstalled = false, $pivot = 'lang_code', $orderField = null,
```
Comment (HLeithner) [inline libraries/src/Language/LanguageHelper.php]: We can't change the default setting without a proper deprecation/change notification. please remove this change.

**Comment by richard67** [comment]
Back to pending due to review comment here: [https://github.com/joomla/joomla-cms/pull/27341#discussion_r361733183](https://github.com/joomla/joomla-cms/pull/27341#discussion_r361733183 "https://github.com/joomla/joomla-cms/pull/27341#discussion_r361733183")<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27341">issues.joomla.org/tracker/joomla-cms/27341</a>.</sub>

**Comment by richard67** [comment]
I have not tested this item.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27341">issues.joomla.org/tracker/joomla-cms/27341</a>.</sub>

File: libraries/src/Language/LanguageHelper.php (hunk 1 above)
Comment (infograf768) [reply libraries/src/Language/LanguageHelper.php]: Then I will just change to false where it should be.

**Comment by infograf768** [comment]
Will change tomorrow

File: libraries/src/Language/LanguageHelper.php (hunk 1 above)
Comment (HLeithner) [reply libraries/src/Language/LanguageHelper.php]: Yes

**Comment by richard67** [comment]
Then I will test tomorrow ;-)

**Comment by infograf768** [comment]
@Fedik @richard67 @HLeithner 
Modified in order to not change the library.

Please test again.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 64fe02e49d66fb0269437a8f1bdc4e0afa666c98<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27341">issues.joomla.org/tracker/joomla-cms/27341</a>.</sub>

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 64fe02e49d66fb0269437a8f1bdc4e0afa666c98<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27341">issues.joomla.org/tracker/joomla-cms/27341</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27341">issues.joomla.org/tracker/joomla-cms/27341</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**